cloud/amazon/ec2_vpc_peer.py

```diff
@@ -0,0 +1,1544 @@
+                        "Value": "env",
+                        "Key": "Management"
+                    },
+                    {
+                        "Value": "Management to Production",
+                        "Key": "Name"
+                    }
+                ],
+                "AccepterVpcInfo": {
+                    "OwnerId": "12345678910",
+                    "VpcId": "vpc-123456789",
+                    "CidrBlock": "172.31.0.0/16"
+                },
+                "VpcPeeringConnectionId": "pcx-12345678",
+                "RequesterVpcInfo": {
+                    "OwnerId": "12345678910",
+                    "VpcId": "vpc-12345678",
+                    "CidrBlock": "172.32.0.0/16"
+                }
+            }
+        ]
+    ]
+    Returns:
+        Tuple (bool, str, list)
+    """
+    success = False
+    err_msg = ''
+    params = {
+        'DryRun': check_mode
+    }
+    result = list()
+    if vpc_id and vpc_peer_id:
+        params['Filters'] = [
+            {
+                'Name': 'requester-vpc-info.vpc-id',
+                'Values': [vpc_id],
+            },
+            {
+                'Name': 'accepter-vpc-info.vpc-id',
+                'Values': [vpc_peer_id],
+            }
+        ]
+        if status_codes:
+            params['Filters'].append(
+                {
+                    'Name': 'status-code',
+                    'Values': status_codes
+                }
+            )
+
+    elif vpc_peering_id:
+        params['VpcPeeringConnectionIds'] = [vpc_peering_id]
+        if status_codes:
+            params['Filters'] = [
+                {
+                    'Name': 'status-code',
+                    'Values': status_codes
+                }
+            ]
+
+    try:
+        result = (
+            client.describe_vpc_peering_connections(**params)
+            ['VpcPeeringConnections']
+        )
+        success = True
+
+    except botocore.exceptions.ClientError, e:
+        if e.response['Error']['Code'] == 'DryRunOperation':
+            success = True
+            err_msg = e.message
+        else:
+            err_msg = str(e)
+
+    return success, err_msg, result
+
+def is_active(peering_conn):
+    return peering_conn['status']['code'] == 'active'
+
+def is_deleted(peering_conn):
+    return peering_conn['status']['code'] == 'deleted'
+
+def is_expired(peering_conn):
+    return peering_conn['status']['code'] == 'expired'
+
+def is_failed(peering_conn):
+    return peering_conn['status']['code'] == 'failed'
+
+def is_initiating_request(peering_conn):
+    return peering_conn['status']['code'] == 'initiating-request'
+
+def is_pending(peering_conn):
+    return peering_conn['status']['code'] == 'pending-acceptance'
+
+def is_provisioning(peering_conn):
+    return peering_conn['status']['code'] == 'provisioning'
+
+def is_rejected(peering_conn):
+    return peering_conn['status']['code'] == 'rejected'
+
+def make_tags_in_proper_format(tags):
```

Comment:
Can this be replaced with the helper function? https://github.com/ansible/ansible-modules-extras/blob/devel/cloud/amazon/GUIDELINES.md#boto3_tag_list_to_ansible_dict